src/lib/components/Segment.svelte

```diff
@@ -0,0 +1,121 @@
+<script lang="ts">
+  import { writable } from "svelte/store";
+  import type { SegmentContext, SelectedSegment } from "$lib/types/segment";
+  import { SEGMENT_CONTEXT_KEY } from "$lib/types/segment";
+  import { setContext } from "svelte";
+
+  export let selectedSegmentId: symbol | undefined = undefined;
+
+  const store = writable<SelectedSegment>({
+    id: selectedSegmentId,
+    element: undefined,
+  });
+
+  setContext<SegmentContext>(SEGMENT_CONTEXT_KEY, {
+    store,
+  });
+
+  $: $store, (() => (selectedSegmentId = $store.id))();
+
+  let indicator:
+    | {
+        left: number;
+        width: number;
+      }
+    | undefined = undefined;
+
+  let segment: HTMLElement | undefined | null;
+
+  let selectedElement: HTMLElement | undefined | null;
+  $: selectedElement =
+    $store.element ?? segment?.querySelector(".segment-button");
+
+  export const initIndicator = () => {
+    if (selectedElement === undefined || selectedElement === null) {
+      indicator = undefined;
+      return;
+    }
+
+    const { parentElement } = selectedElement;
+
+    if (parentElement === null) {
+      indicator = undefined;
+      return;
+    }
+
+    const { left: parentClientLeft } = parentElement.getBoundingClientRect();
+    const { left: currentClientLeft, width: currentClientWidth } =
+      selectedElement.getBoundingClientRect();
+
+    indicator = {
+      left: currentClientLeft - parentClientLeft,
+      width: currentClientWidth,
+    };
+  };
+
+  $: selectedElement, (() => initIndicator())();
+
+  let segmentsCount = 0;
+  $: segment,
+    (() =>
+      (segmentsCount =
```

Comment:
If a button is added dynamically, will it be taken into account?

`segment` variable will be the same HTMLElement; therefore, it won't change

It might not be necessary to solve it in this PR, but it's worth adding a TODO and solving afterwards?

Reply:
Sure, added todo